signac/job.py

```diff
@@ -18,15 +19,38 @@
     json_attr_dict_validator,
 )
 from ._synced_collections.errors import KeyTypeError
+from ._synced_collections.utils import SyncedCollectionJSONEncoder
 from ._utility import _mkdir_p
 from .errors import DestinationExistsError, JobsCorruptedError
 from .h5store import H5StoreManager
-from .hashing import calc_id
 from .sync import sync_jobs
 
 logger = logging.getLogger(__name__)
 
 
+def calc_id(statepoint):
```

Comment:
Would this work better as a method of the job class?

Reply:
A free function is my preference here. Most hash functions are written as free functions, and exceptions that I can think of (like parts of Python's `hashlib`) are related to persisting object state for hashing large inputs.